components/locid_transform/src/provider/canonicalizer.rs

```diff
@@ -79,3 +79,143 @@ pub struct AliasesV1<'data> {
     #[cfg_attr(feature = "serde", serde(borrow))]
     pub subdivision: ZeroMap<'data, UnvalidatedSubdivision, SemivalidatedSubdivision>,
 }
+
+impl<'data> From<AliasesV2<'data>> for AliasesV1<'data> {
+    fn from(value: AliasesV2<'data>) -> Self {
+        let language_variants = value
+            .language_variants
+            .iter()
+            .map(zerofrom::ZeroFrom::zero_from)
+            .map(|v: StrStrStrPair| {
+                let langid = alloc::format!("{0}-{1}", v.0, v.1);
+                StrStrPair(langid.into(), v.2)
+            })
+            .collect::<alloc::vec::Vec<StrStrPair>>();
+
+        Self {
+            language_variants: VarZeroVec::from(&language_variants),
+            sgn_region: value.sgn_region,
+            language_len2: value.language_len2,
+            language_len3: value.language_len3,
+            language: value.language,
+            script: value.script,
+            region_alpha: value.region_alpha,
+            region_num: value.region_num,
+            complex_region: value.complex_region,
+            variant: value.variant,
+            subdivision: value.subdivision,
+        }
+    }
+}
+
+impl<'data> From<AliasesV1<'data>> for AliasesV2<'data> {
+    fn from(value: AliasesV1<'data>) -> Self {
+        #[allow(unused_imports)]
+        use alloc::borrow::ToOwned;
+
+        let language_variants = value
+            .language_variants
+            .iter()
+            .map(zerofrom::ZeroFrom::zero_from)
+            .map(|v: StrStrPair| {
+                #[allow(clippy::expect_used)]
+                let (lang, variant) =
+                    v.0.split_once('-')
+                        .expect("Each pair should be language-variant");
```

Comment:
AliasesV1 could be garbage, you can't expect this.

Reply:
should I throw here?

Reply:
You should change the `expect` to a `debug_assert` and then return a GIGO value in accordance with the style guide (https://github.com/unicode-org/icu4x/blob/main/docs/design/data_safety.md)

Reply:
You can either make this a `TryFrom` and throw in the constructor that uses this, or do garbage-in garbage-out

Reply:
Yeah, TryFrom might be better here.